functions/dialect_test.go

```diff
@@ -564,3 +566,123 @@ func checkCompoundNames(t *testing.T, compoundNames []string, expectedNames []st
 		assert.Contains(t, compoundNames, name)
 	}
 }
+
+func TestScalarFunctionsMatch(t *testing.T) {
```

Comment:
rather than have a mega yaml and many test cases, can we split into little yamls for each test case?

Reply:
splitted it